pennylane/labs/dla/dense_util.py

```diff
@@ -235,6 +236,86 @@ def pauli_decompose(H: TensorLike, tol: Optional[float] = None, pauli: bool = Fa
     return H_ops
 
 
+def orthonormalize(basis):
+    r"""Orthonormalize a list of basis vectors"""
+
+    if isinstance(basis, PauliVSpace) or all(
+        isinstance(op, (PauliSentence, Operator)) for op in basis
+    ):
+        return _orthonormalize_ps(basis)
+
+    if all(isinstance(op, np.ndarray) for op in basis):
+        return _orthonormalize_np(basis)
+
+    raise NotImplementedError(
+        f"orthonormalize not implemented for basis of type {type(basis[0])}:\n{basis}"
+    )
+
+
+def _orthonormalize_np(basis: Iterable[np.ndarray]):
+    basis = np.array(basis)
+    gram_inv = np.linalg.pinv(sqrtm(trace_inner_product(basis, basis).real))
+    return np.tensordot(gram_inv, basis, axes=1)
+
+
+def _orthonormalize_ps(basis: Union[PauliVSpace, Iterable[Union[PauliSentence, Operator]]]):
```

Comment:
Maybe a few code comments could help in this function :) Alternatively, a (private) docstring could give some overview of what is going to happen.

Reply:
✅ good call!